### PR TITLE
COL-679 Pull preview images from middle of video clips

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/COL-679

`w=1` may be helpful for review.

### CODE REVIEW

File: lib/preview/video.js [hunk 1]
```diff
@@ -68,34 +70,58 @@ var process = module.exports.process = function(ctx, callback) {
...
-  exec(extractCommand, {'timeout': 200000}, function(err, stdout, stderr) {
+  exec(getDurationCommand, {'timeout': 30000}, function(err, stdout, stderr) {
+    var timecode = 0;
     if (err) {
-      var msg = util.format('Error while extracting first frame image from video: %s', err.message);
-      return callback({'code': 500, 'msg': msg});
+      log.error({'err': err, 'path': ctx.sourcePath}, 'Error getting video duration, will attempt to use first frame');
+    } else {
+      var duration = Number(stdout);
+      if (Number.isNaN(duration)) {
```
Comment (johncrossman) [inline lib/preview/video.js]: @paulkerschen Forgive my ignorance: will the above `isNaN` behave as expected?  See https://ariya.io/2014/05/the-curious-case-of-javascript-nan  (e.g., `Number.isNaN('hello');  // false`?!)  Am I misreading?

Comment (johncrossman) [inline lib/preview/video.js]: Ah, the previous line makes it clear. I get it. :)

Comment (paulkerschen) [inline lib/preview/video.js]: @johncrossman Not that JavaScript makes this easy, but I think it should work right in combination with the `Number()` cast above exists.

// Number('hello'); // NaN
// Number.isNaN(Number('hello')); // true

**Comment by johncrossman** [comment]
Bravo, @paulkerschen

**Pull request merged**